src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -165,9 +165,9 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 			throw new Error("Too many arguments")
 		}
 		let result;
-		if (args.length === 1) {
+		if (args && args.length === 1) {
```

Comment:
Have you search for all args.length usage in all related files?  I see another possibility in insertMany

Reply:
We need at least one parameter for the [insertMany](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#insertMany) call - reflected in the _mandatory argument array_ to its call. In that case, havign args as undefined would be an error we want to propagate. 
NB: Given [the call](https://github.com/Microsoft/vscode-cosmosdb/blob/b66d52b14e86096809b1bc80ebd8eba3995a396d/src/mongo/tree/MongoCollectionTreeItem.ts#L112), args being undefined would throw an error at the call site itself.

Reply:
I don't quite follow what you're saying. Yes an error is expected, but you need to make sure it's the correct error. For example it should be "Too few arguments" instead of "Cannot access 'length' of undefined". Is that the case currently? I feel like the second error would be displayed, which is not what we want